src/components/DeleteUser.js

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+const DeleteUser = ( { user, password } ) => {
```

Comment:
A user (that is logged in) will click a button "Delete account" and as a result of that action a window will be shown that:
- Will display a message informing the user that this operation is permanent
- Provide an input box for the user to input his password for confirmation

- As props, this will only get the user, because the password the user enters will be passed to the API which will validate it (line 3)
- You don't need to bother with conditionally displaying any information on this form based on whether the user is logged in because the form itself will be available only when the user is logged in. (line 14-16)

Reply:
Okay mate I think I've got this to a passable level. Bare with me. I didn't see these comments yesterday.